tests/healthReadyTests.js

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2021-2022 Inclusive Design Research Centre, OCAD University
+ * All rights reserved.
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ */
+
+"use strict";
+
+require("json5/lib/register");
+const fluid = require("infusion");
+const jqUnit = require("node-jqunit");
+
+require("../src/shared/driverUtils.js");
+require("./shared/utilsCommon.js");
+require("./shared/utilsSso.js");
+
+jqUnit.module("Personal Data Server /health and /ready tests.");
+
+fluid.registerNamespace("fluid.tests.healthReady");
+
+const skipDocker = process.env.SKIPDOCKER === "true" ? true : false;
+const path = require("path");
+const config = require("../src/shared/utils.js").loadConfig(path.join(__dirname, "testConfig.json5"));
+const serverUrl = "http://localhost:" + config.server.port;
+fluid.tests.utils.setDbEnvVars(config.db);
+
+const postgresOps = require("../src/dbOps/postgresOps.js");
+const postgresHandler = new postgresOps.postgresOps(config.db);
+
+const server = require("../server.js");
+
+jqUnit.test("Health and Ready end point tests", async function () {
+    jqUnit.expect(skipDocker ? 16 : 18);
+    let serverStatus, response;
+
+    // Start with server off -- "/health" should fail
+    response = await fluid.tests.utils.sendRequest(serverUrl, "/health");
+    jqUnit.assertNotNull("Check '/health' error", response);
+    jqUnit.assertTrue("Check '/health' error code", response.toString().includes("ECONNREFUSED"));
+
+    // Start server, but not the database.
+    const serverInstance = server.startServer(config.server.port);
+    serverStatus = await fluid.personalData.getServerStatus(config.server.port);
+    jqUnit.assertTrue("The server is up and running", serverStatus);
+
+    // "/health" request should now succeed ...
+    response = await fluid.tests.utils.sendRequest(serverUrl, "/health");
+    fluid.tests.healthReady.testResult(response, 200, { isHealthy: true }, "/health (should succeed)");
+
+    //  ... but "/ready" should fail
+    response = await fluid.tests.utils.sendRequest(serverUrl, "/ready");
+    fluid.tests.healthReady.testResult(response, 503, { isError: true, message: "Database is not ready" }, "/ready (should error)");
```

Comment:
This one is awkward since if the user is running with a local database, it will always be ready.  This test should be skipped if skipDocker is true.